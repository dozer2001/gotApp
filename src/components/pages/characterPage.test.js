import React from 'react';
import {shallow} from 'enzyme';
import CharacterPage from './characterPage';


describe('Testing <CharacterPage/>', () => {
    const character = shallow(<CharacterPage/>);
    describe('Testing  state', () => {
        it('houses have rendered correctly', () => {
            expect(character).toMatchSnapshot();
        });
        it('houses state "houses" is empty obj', () => {
            expect(character.state().selectedChar).toBeNull();
        });

        it('houses state "error" is ', () => {
            expect(character.state().error).toBeFalsy();
        });
    });
    describe('Habdlers test', () => {
        it('testing character', () => {
            character.instance().onItemSelected();
            expect(character.state().error).toBeFalsy();
        });

    });
});