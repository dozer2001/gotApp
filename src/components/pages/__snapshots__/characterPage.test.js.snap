// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CharacterPage/> Testing  state houses have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CharacterPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "left": <ItemList
        getData={[Function]}
        onItemSelected={[Function]}
        renderItem={[Function]}
      />,
      "right": <ItemDetails
        getData={[Function]}
        id={null}
      >
        <Field
          field="gender"
          label="Gender"
        />
        <Field
          field="born"
          label="Born"
        />
        <Field
          field="died"
          label="Died"
        />
        <Field
          field="culture"
          label="Culture"
        />
      </ItemDetails>,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "left": <ItemList
          getData={[Function]}
          onItemSelected={[Function]}
          renderItem={[Function]}
        />,
        "right": <ItemDetails
          getData={[Function]}
          id={null}
        >
          <Field
            field="gender"
            label="Gender"
          />
          <Field
            field="born"
            label="Born"
          />
          <Field
            field="died"
            label="Died"
          />
          <Field
            field="culture"
            label="Culture"
          />
        </ItemDetails>,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing <characterPage/> Testing  state houses have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <characterPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "characterPage",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "characterPage",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
